### PR TITLE
feat(FluidNumberInput): add `FluidNumberInput` variant

### LINKED ISSUE #12454: Fluid inputs: Add FluidNumberInput experimental implementation
https://github.com/carbon-design-system/carbon/issues/12454#tasklist-block-c3f2608b-b432-42df-bb01-c5a50c9a75ac

### LINKED ISSUE #12455: Fluid inputs: FluidNumberInputSkeleton

### LINKED ISSUE #12826: [Bug]: Update number input type token
### Package

@carbon/react

### Browser

_No response_

### Package version

current version

### Description

We want to change number input from using the monospaced `$code` token to just the regular `$body-compact-01`

### Screenshots

image

### Steps to reproduce

Go to https://react.carbondesignsystem.com/?path=/story/components-numberinput--playground

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

### PR DESCRIPTION
**Pull request by tw15egan**
Refs https://github.com/carbon-design-system/carbon/issues/12126
Closes https://github.com/carbon-design-system/carbon/issues/12454
Closes https://github.com/carbon-design-system/carbon/issues/12455
Closes https://github.com/carbon-design-system/carbon/issues/12826

-Adds in the `FluidNumberInput` component as `unstable`

#### Changelog

**New**
- `unstable__FluidNumberInput` component
- Styles for `FluidNumberInput` (`_fluid-number-input.scss`)
- Stories for `FluidNumberInput` (`FluidNumberInput.stories.js`)
- Tests for `FluidNumberInput`

**Changed**

- Updates index, snapshots

#### Testing / Reviewing

Go to `FluidNumberInput` and ensure all stories look correct

### CODE REVIEW

File: packages/react/src/components/FluidNumberInput/FluidNumberInput.stories.js [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+  Toggletip,
+  ToggletipButton,
+  ToggletipContent,
+} from '../Toggletip';
+import { Information } from '@carbon/icons-react';
+
+export default {
+  title: 'Experimental/unstable__FluidNumberInput',
+  component: FluidNumberInput,
+};
```
Comment (tay1orjones) [inline packages/react/src/components/FluidNumberInput/FluidNumberInput.stories.js]: Could `FluidNumberInputSkeleton` be added as a subcomponent so it shows up in the prop api table on the mdx docs?

File: packages/react/src/components/FluidNumberInput/FluidNumberInput.stories.js [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+      type: 'text',
+    },
+    defaultValue:
+      'Warning message that is really long can wrap to more lines but should not be excessively long.',
+  },
+  // value: {
+  //   control: {
+  //     type: 'text',
+  //   },
+  // },
```
Comment (tay1orjones) [inline packages/react/src/components/FluidNumberInput/FluidNumberInput.stories.js]:
```suggestion
```

**Review by tay1orjones — CHANGES_REQUESTED** [review]
This looks super good! A couple minor things below.

Also, I think `packages/styles/scss/components/__tests__/number-input-test.js` (and others - `fluid-combo-box`, `fluid-multiselect`, etc.) need updated to ensure the new mixins exist. Sorry I didn't catch this before - might be something to do for all fluid inputs in a separate PR?

File: packages/react/src/components/FluidNumberInput/FluidNumberInput.stories.js (hunk 1 above)
Comment (tw15egan) [reply packages/react/src/components/FluidNumberInput/FluidNumberInput.stories.js]: Oh good catch, will add that

**Comment by tw15egan** [comment]
I'll tackle that in a separate PR (created issue https://github.com/carbon-design-system/carbon/issues/12817) if that's alright. Pushed up changes adding all skeleton components as subcomponents to the story files

**Review by tay1orjones — APPROVED** [review]

**Review by aagonzales — COMMENTED** [review]
It looks good except for the label overflow (which I keep forgetting about). It needs to overflow like how we set it up in [fluid text input](https://react.carbondesignsystem.com/?path=/story/experimental-unstable-fluidtextinput--playground). But that's the only thing I see wrong.

image

**Comment by tw15egan** [comment]
@aagonzales I'll update this one here, and I've created a new issue (https://github.com/carbon-design-system/carbon/issues/12825) to track the rest of the components and will add the same styles to all of them in a separate PR

**Review by aagonzales — COMMENTED** [review]
That looks good! One more change after talking to Jeannie, we want to just use the regular `$body-compact-01` here instead of the mono `$code-01`. She wants to change it in the fixed version as well.

**Review by alisonjoseph — APPROVED** [review]
LGTM!

**Comment by tw15egan** [comment]
@aagonzales updated 👍🏻

**Comment by aagonzales** [comment]
@tw15egan it does look like the number are appearing in Plex.

image


What the plex number should look like
image

**Comment by tw15egan** [comment]
@aagonzales good catch, forgot number inputs reset their own styles, and it was using the browser default. Updated it, so it should not load in IBM Plex Sans

**Review by aagonzales — APPROVED** [review]
Perfect! Ship it!

**Pull request merged**